### PR TITLE
[CHK-1565] feat(monitoring): correlate event dispatcher async processing with original endpoint

### PR DESCRIPTION
**Pull request by giovanniberti**
**NOTE:** Was #108, mistakenly closed
 

#### List of Changes

* correlate event dispatcher async processing with original endpoint

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/94
Depends on #107

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -125,6 +125,11 @@
 			<artifactId>mockito-inline</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>io.mockk</groupId>
+			<artifactId>mockk-jvm</artifactId>
+			<version>1.13.5</version>
```
Comment (pietro-tota) [inline pom.xml:131]: Add scope test for this library

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt [hunk 2]
```diff
@@ -46,13 +93,19 @@ class TransactionsRefundQueueConsumer(
...
 
-  private fun getTransactionIdFromPayload(data: BinaryData): Mono<String> {
-    val idFromActivatedEvent =
-      data.toObjectAsync(TransactionExpiredEvent::class.java).map { it.transactionId }
-    val idFromRefundedEvent =
-      data.toObjectAsync(TransactionRefundRetriedEvent::class.java).map { it.transactionId }
+  private fun parseEvent(data: BinaryData): Mono<RefundQueueEvent> {
+    val refundRequestedEvent = data.toObjectAsync(TransactionRefundRequestedEvent::class.java)
+    val refundRetriedEvent = data.toObjectAsync(TransactionRefundRetriedEvent::class.java)
+    val expiredEvent = data.toObjectAsync(TransactionExpiredEvent::class.java)
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt:99]: I think that TransactionExpiredEvent should not be sent on refund queue, right?

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt [hunk 3]
```diff
@@ -465,6 +467,10 @@ fun <T> runPipelineWithDeadLetterQueue(
           logger.info(
             "Event: [${eventPayload.toString(StandardCharsets.UTF_8)}] successfully sent with visibility timeout: [${it.value.timeNextVisible}] ms to queue: [${deadLetterQueueAsyncClient.queueName}]")
         }
+        .doOnNext {
+          logger.info(
+            "Event: [${eventPayload.toString(StandardCharsets.UTF_8)}] successfully sent with visibility timeout: [${it.value.timeNextVisible}] ms to queue: [${deadLetterQueueAsyncClient.queueName}]")
+        }
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt:473]: duplicated doOnNext?

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/eventretry/TracedRetryEventService.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        it.status = newStatus
+        viewRepository.save(it).flatMap { Mono.just(event) }
+      }
+
+  private fun enqueueMessage(
+    event: E,
+    visibilityTimeout: Duration,
+    tracingInfo: TracingInfo
+  ): Mono<Void> =
+    Mono.just(event).flatMap { eventToSend ->
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/eventretry/TracedRetryEventService.kt:88]: remove eventToSend unused lamba parameter

**Review by pietro-tota — DISMISSED** [review]
Well done, just some comment

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt (hunk 2 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt:99]: You're right, removing it

Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt]: Fixed in 32d29685af54e9875fd29620a6e07ea8dc76c453

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt (hunk 3 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt]: Fixed in Fixed in 5ceb643730d965324fb2d6321260788bdee1c44b

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/eventretry/TracedRetryEventService.kt (hunk 4 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/eventretry/TracedRetryEventService.kt]: Fixed in 91f58aca494d54719b19658383efb941326ac1be

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt (hunk 2 above)
Comment (giovanniberti) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionsRefundQueueConsumer.kt]: You're right, fixed in 32d2968

File: pom.xml (hunk 1 above)
Comment (giovanniberti) [reply pom.xml]: Removed as it was used in intermediate experimentation :)

**Pull request merged**